src/manifest/dbtProject.ts

```diff
@@ -625,11 +627,11 @@ export class DBTProject implements Disposable {
     if (!node) {
       return [];
     }
-    return this.getColumsOfRelation(
-      node.database,
-      node.schema,
-      node.alias || tableName,
-    );
+    let _tableName = node.alias || tableName;
+    if (node.resource_type === DBTProject.RESOURCE_TYPE_SOURCE) {
+      _tableName = node.identifier;
+    }
+    return this.getColumsOfRelation(node.database, node.schema, _tableName);
```

Comment:
This should only be called for sources. Please adapt it in that way.

Reply:
Ideally the `ResolveReferenceResult` should be split per Source vs node as they don't have same properties.

Reply:
For now, I have directly used `node.identifier`. Will change `ResolveReferenceResult` in future PR